internal/parser/parser_test.go

```diff
@@ -1799,8 +1801,8 @@ func TestSingleStatementParse(t *testing.T) {
 			},
 		},
 		{
-			"DELETE with basic with clause, select stmt with FROM with joinclause's ON and basic cte-table-name",
-			"WITH myTable AS (SELECT * FROM myTable1,myTable2 ON myExpr) DELETE FROM myTable",
+			"DELETE with basic with clause, select stmt with FROM with joinclause with multiple join-clause-parts, join constraint and basic cte-table-name",
```

Comment:
don't change existing tests, add new ones

Reply:
The changes to the old tests are to accomodate for the slice than the single element. 
The test added is indeed a new one, old tests arent removed :D

Reply:
The changes in the old tests are to accomodate for the change of the DS to a slice.
The test indeed is a new one, no old tests were removed :D